### PR TITLE
Changes to Jenkins to add publishing npm packages

### PR DESCRIPTION
**Pull request by rsomers1998**

### CODE REVIEW

**Comment by rsomers1998** [comment]
Need to find a better way to store repo endpoint but review so far would be great

File: Jenkinsfile [hunk 1]
```diff
@@ -95,14 +110,47 @@ pipeline {
...
+                              npm publish generated-openapi
+                           }
+                        }
                      }
 
                      dir('dev.galasa.framework.api.cps') {
                         sh "mvn --settings ${workspace}/settings.xml -Dmaven.repo.local=${workspace}/repository -Dgpg.skip=${GPG_SKIP} -Dgpg.passphrase=$GPG -P ${MAVEN_PROFILE} -B -e -fae --non-recursive ${MAVEN_GOAL}"
+
+                        sh "npm install @openapitools/openapi-generator-cli"
+                        sh "npx @openapitools/openapi-generator-cli generate -i openapi.yaml --skip-validate-spec --generator-name typescript-rxjs -o generated-openapi --additional-properties=npmName=galasa-cps-api,npmRepository=${npmRepository},npmVersion=${npmVersion},snapshot=${npmSnapshot},supportsES6=false,modelPropertyNaming=original"
```
Comment (Michael-Baylis) [inline Jenkinsfile:130]: package name will need to indicate typescript rxjs,   we may be generated more than one "type"

File: Jenkinsfile [hunk 2]
```diff
@@ -10,6 +10,11 @@ pipeline {
       
 //Set some defaults
       def workspace = pwd()
+
+//Set npm package variables
+      def npmVersion = "0.9.0"
+      def npmSnapshot = true
+      def npmRepository = "https://nexus.galasa.dev/repository/npm-development"
```
Comment (Michael-Baylis) [inline Jenkinsfile:17]: put in folder properties

File: Jenkinsfile [hunk 3]
```diff
@@ -95,14 +110,47 @@ pipeline {
 
                      dir('dev.galasa.framework.api.runs') {
                         sh "mvn --settings ${workspace}/settings.xml -Dmaven.repo.local=${workspace}/repository -Dgpg.skip=${GPG_SKIP} -Dgpg.passphrase=$GPG -P ${MAVEN_PROFILE} -B -e -fae --non-recursive ${MAVEN_GOAL}"
+
+                        sh "npm install @openapitools/openapi-generator-cli"
+                        sh "npx @openapitools/openapi-generator-cli generate -i openapi.yaml --skip-validate-spec --generator-name typescript-rxjs -o generated-openapi --additional-properties=npmName=galasa-runs-api,npmRepository=${npmRepository},npmVersion=${npmVersion},snapshot=${npmSnapshot},supportsES6=false,modelPropertyNaming=original"
+                        sh "npm install --prefix generated-openapi"
+                        script {
+                           if (env.JOB_NAME.contains('PullRequest')) {
```
Comment (Michael-Baylis) [inline Jenkinsfile:118]: put a flag in folder properties.   cant rely on the jobname

**Comment by rsomers1998** [comment]
Found folder props, set PULL_REQ (default false but true for PRs), NPM_VERSION, NPM_SNAPSHOT, NPM_REPO

**Pull request merged**